File: chessgame-api/src/main/java/git/dimitrikvirik/chessgameapi/model/domain/GameHistory.java
package git.dimitrikvirik.chessgameapi.model.domain;


import git.dimitrikvirik.chessgameapi.model.game.Action;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@Entity
public  class GameHistory {
    @Id
    @GeneratedValue(
            strategy = GenerationType.IDENTITY
    )
    private Integer id;
    private LocalDateTime time;
    @Enumerated
    private Action action;
    private Integer from_x;
    private Integer from_y;
    private Integer to_x;
    private Integer to_y;
    @ManyToOne
    private Game game;
}

File: chessgame-api/src/main/kotlin/git/dimitrikvirik/chessgameapi/model/domain/GameHistory.kt
package git.dimitrikvirik.chessgameapi.model.domain

import git.dimitrikvirik.chessgameapi.model.game.Action
import org.hibernate.Hibernate
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "game_history")
class GameHistory(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    var time: LocalDateTime? = null,
    @Enumerated var action: Action? = null,
    var from_x: Int? = null,
    var from_y: Int? = null,
    var to_x: Int? = null,
    var to_y: Int? = null,
    @ManyToOne
    var game: Game? = null,
)